### PR TITLE
Add guile-sdl2 package to nixpkgs

### PR DESCRIPTION
**Pull request by seppeljordan**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Reguarding sandboxing: I cannot find it in the manual link you provided, sorry.

### CODE REVIEW

**Comment by joachifm** [comment]
If you're on an older version of NixOS, the option you want is `nix.useChroot`

**Comment by zimbatm** [comment]
To fit the CONTRIBUTING the commit should be named: `guile-sdl2: init at 0.1.0`

File: pkgs/development/guile-modules/guile-sdl2/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                  SDL2
+                  SDL2_image
+                  SDL2_ttf
+                  SDL2_mixer
+                  libtool
+                  guile
+                  pkgconfig ];
+  src = fetchgit {
+    url = "git://dthompson.us/guile-sdl2.git";
+    rev = "048f80ddb5c6b03b87bba199a99a6f22d911bfff";
```
Comment (zimbatm) [inline pkgs/development/guile-modules/guile-sdl2/default.nix]: you can also use a git tag as a reference: `rev = "v${version}";`

File: pkgs/development/guile-modules/guile-sdl2/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  '';
+  configureFlags = [ "--with-libsdl2-prefix=${SDL2}"
+                     "--with-libsdl2-image-prefix=${SDL2_image}"
+                     "--with-libsdl2-ttf-prefix=${SDL2_ttf}"
+                     "--with-libsdl2-mixer-prefix=${SDL2_mixer}"];
+  makeFlags = ["GUILE_AUTO_COMPILE=0"];
+  meta = {
+    description = "Bindings to SDL2 for GNU Guile";
+    homepage = "https://git.dthompson.us/guile-sdl2.git";
+    license = stdenv.lib.licenses.gpl3;
```
Comment (zimbatm) [inline pkgs/development/guile-modules/guile-sdl2/default.nix]: do you plan on maintaining the package ? if yes you can add yourself as the maintainer

Comment (zimbatm) [inline pkgs/development/guile-modules/guile-sdl2/default.nix]: if you want the package to be built by hydra you should also add the `platforms` attribute

**Comment by seppeljordan** [comment]
@zimbatm sorry i am missing something, but isn't that what i named my commit?

**Comment by zimbatm** [comment]
my bad, i assumed the PR title was the same as the commit name

**Comment by seppeljordan** [comment]
@zimbatm would it be okay if i amend the commit, or should i make a new one to incorporate your suggestions?

**Comment by zimbatm** [comment]
amending is cool

**Comment by seppeljordan** [comment]
I thinks it is ready now.

File: lib/maintainers.nix [hunk 3]
```diff
@@ -328,6 +328,7 @@
   schmitthenner = "Fabian Schmitthenner <development@schmitthenner.eu>";
   schristo = "Scott Christopher <schristopher@konputa.com>";
   scolobb = "Sergiu Ivanov <sivanov@colimite.fr>";
+  sebastianjordan = "Sebastian Jordan <sebastian.jordan.mail@googlemail.com>";
```
Comment (zimbatm) [inline lib/maintainers.nix]: the left hand side should be your github handle. the idea is to make it easier to contact you when an issue is opened against the package.

**Comment by seppeljordan** [comment]
changed my entry in maintainers.nix to conform the standards.

**Comment by zimbatm** [comment]
Perfect, thanks !

**Pull request merged**